client/app/cloud/project/openstack/users/add/openstack-users-add.controller.js

```diff
@@ -1,37 +1,39 @@
 class CloudProjectOpenStackUserAddCtrl {
-    constructor ($translate, $uibModalInstance, CloudMessage, OpenstackUsersPassword, OvhApiCloud, serviceName) {
+    constructor ($translate, $uibModalInstance, ControllerHelper, CloudMessage, OpenstackUsersPassword, OvhApiCloud, serviceName) {
         this.$translate = $translate;
         this.$uibModalInstance = $uibModalInstance;
+        this.ControllerHelper = ControllerHelper;
         this.CloudMessage = CloudMessage;
         this.OpenstackUsersPassword = OpenstackUsersPassword;
         this.OvhApiCloud = OvhApiCloud;
         this.serviceName = serviceName;
 
         this.model = {
             value: undefined
-        }
+        };
     }
 
     cancel () {
         this.$uibModalInstance.dismiss();
     }
 
     confirm () {
-        this.saving = true;
-        return this.OvhApiCloud.Project().User().v6().save({
-            serviceName: this.serviceName
-        }, {
-            description: this.model.value
-        }).$promise
-            .then(newUser => {
-                this.OpenstackUsersPassword.put(this.serviceName, newUser.id, newUser.password);
-                this.CloudMessage.success(this.$translate.instant("openstackusers_users_userlist_add_submit_success"));
-            })
-            .catch(err => this.CloudMessage.error([this.$translate.instant("openstackusers_users_userlist_add_submit_error"), err.data && err.data.message || ""].join(" ")))
-            .finally(() => {
-                self.saving = false;
-                this.$uibModalInstance.close();
-            });
+        this.CloudMessage.flushChildMessage();
+        this.saving = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () =>
+                this.OvhApiCloud.Project().User().v6().save({
+                    serviceName: this.serviceName
+                }, {
+                    description: this.model.value
+                }).$promise
+                    .then(newUser => {
+                        this.OpenstackUsersPassword.put(this.serviceName, newUser.id, newUser.password);
+                        this.CloudMessage.success(this.$translate.instant("openstackusers_users_userlist_add_submit_success"));
+                    })
+                    .catch(err => this.CloudMessage.error([this.$translate.instant("openstackusers_users_userlist_add_submit_error"), err.data && err.data.message || ""].join(" ")))
```

Comment:
You can pass an argument in translation like `<translation> Lorem ipsum {{ myParam }}</translation>` and then directly call `this.$translate.instant(id, {myParam})`

Reply:
Your point is very valid but the scope of this work is to replace cui-modal with oui-modal. Your suggestion needs changes in all language translation files and it takes more time. Currently, we have not accounted for this in the current sprint. We can have a different jira story for this kind of changes and take it in next sprint.